frontend/src/App.jsx

```diff
@@ -409,21 +412,39 @@ class App extends PureComponent {
    * Applies a list of deltas to the elements.
    */
   handleDeltaMsg = (deltaMsg, metadataMsg) => {
-    // (BUG #685) When user presses stop, stop adding elements to
-    // report immediately to avoid race condition.
-    // The one exception is static connections, which do not depend on
-    // the report state (and don't have a stop button).
-    const isStaticConnection = this.connectionManager.isStaticConnection()
-    const reportIsRunning =
-      this.state.reportRunState === ReportRunState.RUNNING
-    if (isStaticConnection || reportIsRunning) {
-      this.setState(state => ({
-        // Create brand new `elements` instance, so components that depend on
-        // this for re-rendering catch the change.
-        elements: {
-          ...applyDelta(state.elements, state.reportId, deltaMsg, metadataMsg),
-        },
-      }))
+    this.elementsWithDeltaApplied = applyDelta(
+      this.state.elements,
+      this.state.reportId,
+      deltaMsg,
+      metadataMsg
+    )
+
+    if (!this.timerSet) {
```

Comment:
Use off-the-shelf debounce function, if possible.

Should be the kind of debounce that receives N function calls within 10ms and only files the 1st function call at the end of the 10ms.

Reply:
@tvst Seeing different behavior when using off the shelf `lodash.throttle` function. Have to set the `wait` to 100ms for comparable behavior. Both `hello` and `long_write` examples seem OK at that setting though.